File: core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/promisehandlers/OpenExecutorHandler.scala
package edu.uci.ics.amber.engine.architecture.worker.promisehandlers

import com.twitter.util.Future
import edu.uci.ics.amber.engine.architecture.rpc.controlcommands.{AsyncRPCContext, EmptyRequest}
import edu.uci.ics.amber.engine.architecture.rpc.controlreturns.EmptyReturn
import edu.uci.ics.amber.engine.architecture.worker.DataProcessorRPCHandlerInitializer

trait OpenExecutorHandler {
  this: DataProcessorRPCHandlerInitializer =>

  override def openExecutor(request: EmptyRequest, ctx: AsyncRPCContext): Future[EmptyReturn] = {
    dp.executor.open()
    EmptyReturn()
  }

}

File: core/amber/src/main/python/core/architecture/handlers/control/open_executor_handler.py
from core.architecture.handlers.control.control_handler_base import ControlHandler
from proto.edu.uci.ics.amber.engine.architecture.rpc import EmptyReturn, EmptyRequest


class OpenExecutorHandler(ControlHandler):

    async def open_executor(self, req: EmptyRequest) -> EmptyReturn:
        self.context.executor_manager.executor.open()
        return EmptyReturn()
